src/endpoint/s3/s3_utils.js

```diff
@@ -131,9 +131,19 @@ function get_request_xattr(req) {
 }
 
 function set_response_xattr(res, xattr) {
-    _.each(xattr, (val, key) => {
-        res.setHeader(X_AMZ_META + key, val);
-    });
+    let size_for_md_left = config.S3_MD_SIZE_LIMIT;
+    const sorted_keys = Object.keys(xattr).sort();
```

Comment:
I just noticed that you added this sort in our generic s3 utils.
Not sure this is a good thing to do for all of our namespaces.
I think the size limit is ok, but sorting sounds risky.
Not sure what to suggest, perhaps we can set a special Symbol('XATTR_SORT') on the xattr object in namespace_fs and here we will check for this symbol existence...